File: app/services/accounts.py
from database import supabase
from datetime import datetime

def create_account(name: str, type: str, balance: float = 0):
    data = {
        "name": name,
        "type": type,
        "balance": balance,
        "created_at": datetime.now().isoformat()
    }

    response = supabase.table("accounts").insert(data).execute()
    return response

def register_account():
    print("\n--- Register the new account ---")
    name = input("Account name: ")
    acc_type = input("Account type: ")
    balance = float(input("Initial balance: "))


    print(f"Account '{name}' created successfully!")
    return create_account(name, acc_type, balance)




def get_account(account_id: int):
    response = (
        supabase
        .table("accounts")
        .select("*")
        .eq("id", account_id)
        .single()
        .execute()
    )

    return response.data

File: frontend/src/api/accounts.js
import api from './client'

export const listAccounts = async () => {
  const response = await api.get('/accounts/')
  return response.data
}

export const createAccount = async (name, type, balance) => {
  const response = await api.post('/accounts/', { name, type, balance })
  return response.data
}

export const deleteAccount = async (id) => {
  await api.delete(`/accounts/${id}`)
}